dm/checker/checker.go

```diff
@@ -393,7 +389,7 @@ func (c *Checker) Init(ctx context.Context) (err error) {
 	// Because the table schema obtained from `show create table` is not the schema at the point of binlog.
 	_, checkingShardID := c.checkingItems[config.ShardAutoIncrementIDChecking]
 	_, checkingShard := c.checkingItems[config.ShardTableSchemaChecking]
-	if checkingShard && instance.cfg.ShardMode != "" && instance.cfg.Mode != config.ModeIncrement {
+	if checkingShard && instance.cfg.ShardMode != "" && config.HasLoad(instance.cfg.Mode) {
```

Comment:
I'm not sure, just remember below checkers checks that current upstream table structure can be handled by sharding strategy. So here we should check HasDump & HasLoad?